libraries/ESP8266WiFi/src/ESP8266WiFiSTA.cpp

```diff
@@ -113,21 +113,22 @@ wl_status_t ESP8266WiFiSTAClass::begin(const char* ssid, const char *passphrase,
         return WL_CONNECT_FAILED;
     }
 
-    if(passphrase && strlen(passphrase) > 64) {
+    int passphrase_len = passphrase == NULL ? 0 : strlen(passphrase);
+    if(passphrase_len > 64) {
         // fail passphrase too long!
         return WL_CONNECT_FAILED;
     }
 
     struct station_config conf;
+    conf.threshold.authmode = passphrase_len == 0 ? AUTH_OPEN : _useInsecureWEP ? AUTH_WEP : AUTH_WPA_PSK;

```

Comment:
This line is a bit difficult to understand. Please add parenthesis around the last part: `(_useInsecureWEP ? AUTH_WEP : AUTH_WPA_PSK)`

Reply:
ok